k8s-operator/sessionrecording/conn/conn.go

```diff
@@ -1,20 +0,0 @@
-// Copyright (c) Tailscale Inc & AUTHORS
-// SPDX-License-Identifier: BSD-3-Clause
-
-//go:build !plan9
-
-// Package conn contains shared interface for the hijacked
-// connection of a 'kubectl exec' session that is being recorded.
-package conn
-
-import "net"
-
-type Conn interface {
-	net.Conn
-	// Fail can be called to set connection state to failed. By default any
-	// bytes left over in write buffer are forwarded to the intended
-	// destination when the connection is  being closed except for when the
-	// connection state is failed- so set the state to failed when erroring
-	// out and failure policy is to fail closed.
-	Fail()
```

Comment:
The only reason for this interface to exist was so that we could set the connection status to failed and thus determine if any bytes left in buffer should be send when conn is closing. I was not observe no-written bytes left in buffer on closing, so removed the bit that was conditionally send them